tests/unit_tests/test_openai_embedding_function.py

```
@@ -0,0 +1,355 @@
+    reason="OPENAI_API_KEY environment variable must be set"
+)
+class TestOpenAIEmbeddingFunction:
+    """Test OpenAIEmbeddingFunction - skipped by default, requires manual execution"""
+
+    def test_openai_env(self):
+        """Test if openai package is installed and required environment variables are set."""
+        try:
+            import openai
+        except ImportError:
+            print("openai package is not installed")
+            assert False, "openai package is not installed"
```

Ruff (0.14.11)

Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

Replace `assert False`

(B011)